frame/ranked-collective/src/benchmarking.rs

```diff
@@ -31,20 +31,20 @@ fn assert_last_event<T: Config<I>, I: 'static>(generic_event: <T as Config<I>>::
 	frame_system::Pallet::<T>::assert_last_event(generic_event.into());
 }
 
-fn make_member<T: Config<I>, I: 'static>(rank: Rank) -> Result<T::AccountId, BenchmarkError> {
+fn make_member<T: Config<I>, I: 'static>(rank: Rank) -> T::AccountId {
 	let who = account::<T::AccountId>("member", MemberCount::<T, I>::get(0), SEED);
 	let who_lookup = T::Lookup::unlookup(who.clone());
 	assert_ok!(Pallet::<T, I>::add_member(
-		T::PromoteOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?,
+		T::PromoteOrigin::try_successful_origin().unwrap(),
```

Comment:
So the benchmark now requires there to be a valid `PromoteOrigin`?   
I guess for some basic origins its fine, but for the optional origins we should abort as `Weighless` since otherwise we would force the chain developer to specify an origin although they dont want.

Reply:
Yes, I see the point. 
Both options has pros/cons. 
Aborting with `Weighless` here will set weight to 0 for many extrinsics which do not require the `PromoteOrigin` itself. Which will be wrong, if for example a developer wants temporary disable a promotion only.
Probably a better solution will be to not make a benchmark depend on other (not its direct origin guard) origins.

Reply:
Ah right, this is in a helper function. Yea just wanted to point it out in general, not a big problem.

Reply:
But can we then have an expect with a proper description of the error?

Reply:
@bkchr done!